dags/utility/utility_index_from_s3.py

```
@@ -0,0 +1,188 @@
+    "start_date": datetime(2020, 6, 14),
+    "email": ["systems@digitalearthafrica.org"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=5),
+    "env_vars": {
+        "DB_HOSTNAME": DB_WRITER,
+        "DB_DATABASE": DB_DATABASE,
+        "DB_PORT": DB_PORT,
+        "AWS_DEFAULT_REGION": REGION,
+    },
+    # Lift secrets into environment variables
+    "secrets": [
+        Secret("env", "DB_USERNAME", SECRET_ODC_WRITER_NAME, "postgres-username"),
+        Secret("env", "DB_PASSWORD", SECRET_ODC_WRITER_NAME, "postgres-password"),
+    ],
+}
+
+
+def parse_dagrun_conf(products, **kwargs):
+    """
+    parse input
+    """
+    return products
+
+
+def check_dagrun_config(product_definition_uri: str, s3_glob: str, **kwargs):
+    """
+    determine task needed to perform
+    """
+    logging.info(
+        f"check_dagrun_config - "
+        f"product_definition_uri:{product_definition_uri} - "
+        f"s3_glob:{s3_glob} - "
+        f"kwargs:{kwargs}"
+    )
+
+    if product_definition_uri and s3_glob:
+        return ADD_PRODUCT_TASK_ID
+    elif not product_definition_uri and s3_glob:
+        return INDEXING_TASK_ID
+    else:
+        raise ValueError('Neither product_definition_uri nor s3_glob was informed!')
+
+
+SET_REFRESH_PRODUCT_TASK_NAME = "parse_dagrun_conf"
+CHECK_DAGRUN_CONFIG = "check_dagrun_config"
+
+
+# THE DAG
+dag = DAG(
+    dag_id=DAG_NAME,
+    doc_md=__doc__,
+    default_args=DEFAULT_ARGS,
+    schedule_interval=None,
+    catchup=False,
+    tags=["k8s", "add-product", "self-service", "index-datasets", "explorer-update"],
+)
+
+with dag:
+    TASK_PLANNER = BranchPythonOperator(
+        task_id=CHECK_DAGRUN_CONFIG,
+        python_callable=check_dagrun_config,
+        op_args=[
+            "{{ dag_run.conf.product_definition_uri }}",
+            "{{ dag_run.conf.s3_glob }}",
+        ],
+    )
+
+    ADD_PRODUCT = KubernetesPodOperator(
```

what happen if product already in the database?

---

Silently succeeds still!

---

@alexgleith does is also update product definition if require?